### ISSUE TITLE
Consider adding/setting CMake compiler from conan settings

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#271]
Instead of having users to remember setting CC and CXX environment variables. 
Just an idea for discussion. We have rejected it in the past for doing more than some users wanted, but maybe a second thought can give new ideas.

**Comment by tru** [comment]
Here is my use case for it:

I would like to be able to configure "profiles" similar to what is done with compilers, arch etc in a config file and ask conan to use that profile. So for example:

```
[linux-arm-marvell-gnueabi]
compiler_c=/usr/local/build-envs/linux-marvell-gnueabi/bin/linux-marvell-gnueabi-gcc
compiler_cxx=/usr/...
cflags=-march=armv7h --sysroot=/...
```

This would allow something like
`conan install --target=linux-arm-marvell-gnueabi`

right now we would need a shell script to invoke conan with the correct CC/CXX/CFLAGS environments. I feel like a way to define this in the conan configuration would be much more maintainable.

**Comment by memsharded** [comment]
I like the approach for the compiler paths and other non ABI-modifiers. Doable.
About including the flags in the profile, I think it might not be a good idea. As the binary hash lose that information, and maintaining a portable and common configuration for those binary hashes requires it. 
Put in other words, you will have to manually share your "profiles" with the people you want to be able to consume the packages created with --target=your-target.

I think that @astralien3000 is doing a good job at https://github.com/conan-io/conan/issues/86 and https://github.com/conan-io/conan/pull/294, and our efforts should be oriented in creating some consensus about useful settings. What do you think?

**Comment by memsharded** [comment]
An UX feedback from @sunsided in https://github.com/conan-io/conan/issues/475, pointed again to the existence of "profiles", like: 

```
VSdebug: "-s arch=x86 -s build_type=Debug -s compiler="Visual Studio" -s compiler.runtime=MDd"
VSrelease: "-s arch=x86 -s build_type=Release -s compiler="Visual Studio" -s compiler.runtime=MD"
MinGWrelease: "-s arch=x86 -s build_type=Release -s compiler="gcc"
...
```

Which I think it is closely related to this issue, and we can track progress here, closing the other issue.

**Comment by memsharded** [comment]
This issue might have properly been addressed with the "profiles" feature.
The proposed solution with "profiles" is to allow defining environment variables directly, which seems to be enough to define the compilers paths and flags. 

Labeling as fixed, but feedback appreciated!

**Comment by memsharded** [comment]
the compiler can now be set with environment variables, defined in profiles, or even in command line.
Released in 0.16
